sentinelsat/sentinel.py

```diff
@@ -145,13 +144,13 @@ def query(self, area, initial_date=None, end_date=datetime.now(), **keywords):
         and any other search keywords accepted by the SciHub API.
         """
         query = self.format_query(area, initial_date, end_date, **keywords)
-        self.load_query(query)
-        return self.products
+        return self.load_query(query)
 
-    def load_query(self, query, start_row=0):
+    def load_query(self, query, start_row=0, products=[]):
         """Do a full-text query on the SciHub API using the format specified in
            https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/3FullTextSearch
         """
+        output = products
```

Comment:
I think it might be because your `products` list is not recursively updated when passed through the nested `self.load_query` calls. A NumPy array would. Try `products = self.load_query(...` in [line 194](https://github.com/ibamacsr/sentinelsat/pull/81/files/bef08af026834454d4118ffb9028920c89b72634#diff-b41f2cfc13d00858923dd3e8da963187R194).